src/Transport/HttpTransport.php

```diff
@@ -148,13 +146,19 @@ public function close(?int $timeout = null): PromiseInterface
      */
     private function cleanupPendingRequests(): void
     {
-        while ($promise = array_pop($this->pendingRequests)) {
-            try {
-                $promise->wait();
-            } catch (\Throwable $exception) {
-                // Do nothing because we don't want to break applications while
-                // trying to send events
-            }
+        try {
+            $requestGenerator = function () {
+                foreach ($this->pendingRequests as $key => $request) {
+                    yield $key => $this->httpClient->sendAsyncRequest($request);
+                }
```

Comment:
Can we simplify this with a `yield from $this->pendingRequests`?

Reply:
You can't, we are using the values stored in `$pendingRequests` to pass them to the `sendAsyncRequest` method and not just yielding them

Reply:
Oh cool, I didn't know about `yield from` ;-) ... but yes, in this case it does not help :-)